Bottom: 00d2d7b50396c51da19e212b5b37b693775ca1e4
Top:    31d7ebb1846fb13e6388fe131b598171d473d00a
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-06-29 15:35:10 +0800

Refresh of net-introduce-skb_get_swhash

---

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index aaab06e..f769543 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -930,6 +930,8 @@ static inline __u32 skb_get_hash(struct sk_buff *skb)
 {
 	if (!skb->l4_hash)
 		return skb_get_sw_hash(skb);
+        else if (skb->sw_hash)
+		return skb->sw_hash;
 
 	return skb->hash;
 }
